### DISCUSSION TITLE
Contributing offline support

### DISCUSSION THREAD

**Discussion by mcox-supplyone** [AhashSritharan/Xrm-Ex#7] [Show and tell]
Hi, thanks so much for creating this project! I'm not sure how popular it is--but It's saved me countless hours when working on Dynamics CRM client scripts. At my job I really needed better offline handling as most of my scripts run on the Field Service mobile app in my organization in offline-first mode. I forked this project and added the features I needed, which includes offline detection and an update method for the LookupField. My contribution is pretty basic, but I'd be happy to create a pull request so you can check it out, and maybe it will live on. This would be my first ever contribution to an open-source project, so I wanted to ask before I continue. Thank you.

**Comment by AhashSritharan** [comment]
Hi there!

Thank you so much for your kind words! 😊 I'm thrilled to hear that the project has been helpful for your work. As with most open-source projects, contributions are always welcome, and I’d be happy to review your pull request. Feel free to submit it whenever you're ready!

Looking forward to seeing your contributions!

**Reply by mgcox2** [reply]
So sorry for the delay! I've been slammed at work developing our field service mobile implementation. I can add that my fork of XrmEx has proven itself in the field with our nationwide team of service technicians with offline support. It's worked so well I haven't needed to revisit the library since August! I need to reengage and figure out where I left off so I can make the pull request. Thanks.

**Reply by AhashSritharan** [reply]
There is no rush at all.
Feel free to make the pull request whenever you're ready.
Looking forward to it!
